### PR TITLE
Added global exception handling and report sending to Sentry.io

### PR DESCRIPTION
**Pull request by fwild**
Added exception handling, taking care of #1696

### CODE REVIEW

File: Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   {
+
+       // the Sentry instance
+       public SentrySdk sentry; 
+
+       void Awake()
+       {
+           AppLog.Log("Installing hook for exceptions", LogLevel.INFO );
+           sentry ??= new GameObject("ExceptionManagerSentry").AddComponent<SentrySdk>();
+           sentry.Dsn = "https://b23911205078e7a81bf1489e8aa0fabe@o4506320008118272.ingest.sentry.io/4506320009428992"; // our DSN
```
Comment (cawke) [inline Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs]: move to `const`

File: Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   /// Null Reference, Divide by Zero, Out of Memory, Index Out of Range.
+   /// </summary>
+
+   public class ExceptionManager : MonoBehaviour
+   {
+
+       // the Sentry instance
+       public SentrySdk sentry; 
+
+       void Awake()
```
Comment (cawke) [inline Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs]:
```suggestion
       private void Awake()
```

File: Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+       {
+           AppLog.Log("Installing hook for exceptions", LogLevel.INFO );
+           sentry ??= new GameObject("ExceptionManagerSentry").AddComponent<SentrySdk>();
+           sentry.Dsn = "https://b23911205078e7a81bf1489e8aa0fabe@o4506320008118272.ingest.sentry.io/4506320009428992"; // our DSN
+
+           Application.logMessageReceived += LogCaughtException;
+           DontDestroyOnLoad(gameObject); 
+       }
+
+       void LogCaughtException(string logText, string stackTrace, LogType logType)
```
Comment (cawke) [inline Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs]:
```suggestion
       private void LogCaughtException(string logText, string stackTrace, LogType logType)
```

**Review by cawke — CHANGES_REQUESTED** [review]

File: Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+       {
+
+           if (logType == LogType.Exception)
+           {
+
+              AppLog.Log("ExceptionManager: ["+logType+"]" + logText + ", trace: " + stackTrace, LogLevel.CRITICAL);
+              SentrySdk.CaptureMessage("ExceptionManager: ["+logType+"]" + logText + ", trace: " + stackTrace);
```
Comment (cawke) [inline Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs]:
```suggestion
        AppLog.Log($"ExceptionManager: [{logType}] {logText}, trace: {stackTrace}", LogLevel.CRITICAL);
        SentrySdk.CaptureMessage($"ExceptionManager: [{logType}] {logText}, trace: {stackTrace}");
```

**Review by cawke — CHANGES_REQUESTED** [review]

File: Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs (hunk 1 above)
Comment (fwild) [reply Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs]: can do the next couple of days, but would propose we merge this in now, so we get improved tracking with version 2.5?

Comment (fwild) [reply Assets/MirageXR/Managers/ExceptionManager/ExceptionManager.cs]: @cawke agree?

**Pull request merged**